src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -268,3 +311,27 @@ function validate_connectivity(M, nodes, bus_lookup)
     end
     return connected
 end
+
+# this function extends the PowerModels.jl implementation to accept a System
+function calc_connected_components(sys::System)
+    a = Adjacency(sys; check_connectivity = false)
+    return calc_connected_components(a.data, a.lookup[1])
+end
+
+# this function extends the PowerModels.jl implementation to accept an adjacency matrix and bus lookup
+function calc_connected_components(
```

Comment:
Let's make this function internal because otherwise it will be exported at the top. 

```suggestion
function _calc_connected_components(
```